File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Roots.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;

import org.jetbrains.annotations.NotNull;

public class Roots extends FlavourBuff {
	
	@Override
	public boolean attachTo(@NotNull Char target ) {
		if (target.isFlying()) {
			return false;
		}

        return super.attachTo(target);
	}

	@Override
	public int icon() {
		return BuffIndicator.ROOTS;
	}

	@Override
	public void attachVisual() {
        target.showStatus(CharSprite.NEGATIVE, StringsManager.getVar(R.string.Char_StaRooted));
	}
}

File: scripts/buffs/Roots.lua
---
--- Roots: pins the target in place (was actors/buffs/Roots.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 11, -- BuffIndicator.ROOTS
            name          = "RootsBuff_Name",
            info          = "RootsBuff_Info",
        }
    end,

    attachTo = function(self, buff, target)
        return not target:isFlying()
    end,

    attachVisual = function(self, buff)
        buff.target:showStatus(CharSprite.NEGATIVE, RPD.textById("Char_StaRooted"))
    end
}
